### PR TITLE
Feature: Dark mode support

### PR DESCRIPTION
**Pull request by gangadharbenni**
PR Fixes the Issue : #4 

changes:
- Darkmode config for Tailwind
- Dark mode Toggle button
- Topbar component to add Darkmode Toggler
- Add dark mode contrast color in relevant components

Screenshot (865)

Screenshot (866)

### CODE REVIEW

**Comment by rupali-codes** [comment]
please attack an issue and also follow the [CONTRIBUTING guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)

**Comment by rupali-codes** [comment]
@gangadharbenni its not working, i'm not able see any effects. Can you record a video instead?

**Comment by gangadharbenni** [comment]
@rupali-codes I have linked the issue let me know for any more changes..

File: components/TopBar/TopBar.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { ThemeToggler } from "../ThemeToggler/themeToggler"
+
+export const TopBar = () => {
+    return (
+        <div className="flex justify-end">
+            <span className="mx-10 py-5">
```
Comment (rupali-codes) [inline components/TopBar/TopBar.tsx:6]: increase icon size by `2xl`

**Review by rupali-codes — CHANGES_REQUESTED** [review]
- change title color in `LinksContainer` based on the theme, currently its white for all.

image

- also change the border color of sidenav elements. you can use `border-[rgba(0,0,0,0.3)]`

image
 
- it's not showing anywhere in mobile view, make sure it's located inside the menu

image

**Comment by rohitdasu** [comment]
Screenshot 2023-01-20 at 10 08 46 PM
could you check why light mode is not showing up? maybe for browser settings?

**Comment by gangadharbenni** [comment]
> Screenshot 2023-01-20 at 10 08 46 PM
> 
> could you check why light mode is not showing up? maybe for browser settings?


Looks like its issue because of Daisy UI plugin.

https://github.com/saadeghi/daisyui/discussions/640

**Comment by gangadharbenni** [comment]
@rohitdasu  pls try now made fix for plugin issue..

**Comment by rupali-codes** [comment]
@gangadharbenni i'm not able see logo, kindly fix
image

Thank you

**Comment by anupam42** [comment]
> @gangadharbenni i'm not able see logo, kindly fix image
> 
> Thank you

I think i found the issue

File: components/SideNavbar/SideNavbarElement.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ export const SideNavbarElement = ({ name, url }: SubCategories) => {
   return (
     <Link
       href={url}
-      className={`${ router.asPath === url ? 'bg-slate-300 text-slate-700 transition-all pl-5' : ''} collapse py-3 w-full text-start border-b border-base-100 hover:bg-slate-300 hover:pl-5 hover:text-slate-700 transition-all duration-300 rounded`}
+      className={`${ router.asPath === url ? 'bg-slate-300 text-slate-700 transition-all pl-5' : ''} collapse py-3 w-full text-start border-b border-base-100 hover:bg-slate-300 hover:pl-5 hover:text-slate-700 transition-all duration-300 rounded dark:border-[rgba(0,0,0,0.3)]`}
```
Comment (rupali-codes) [inline components/SideNavbar/SideNavbarElement.tsx:21]: add `border-gray-600`

**Review by rupali-codes — COMMENTED** [review]
make it dark by default, 
and also update this logo (Links) color based on the theme chosen
image

File: components/SideNavbar/SideNavbarElement.tsx (hunk 2 above)
Comment (gangadharbenni) [reply components/SideNavbar/SideNavbarElement.tsx]: updated

**Comment by gangadharbenni** [comment]
> make it dark by default, and also update this logo (Links) color based on the theme chosen image

@rupali-codes
Updated Dark as default on very first visit of site.
There on for next load of site it will use previous set theme in local storage..

Regarding logo color change my thought is like logo is theme independent
hence it should be same between mode changes..

**Review by rohitdasu — APPROVED** [review]

**Comment by rupali-codes** [comment]
@gangadharbenni thanks for quick changes, we appreciate your time and efforts. keep contributing :)

**Review by rupali-codes — APPROVED** [review]

**Pull request merged**